### ISSUE TITLE
[bug][Conan 2.0] Meson generator can't find Ninja

### ISSUE THREAD

**Issue by mkmkme** [conan-io/conan#13344]
### Environment details

* Operating System+version: Debian 11
* Compiler+version: Any
* Conan version: 2.0.1
* Python version: 3.9.2

### Steps to reproduce

```
❯ conan install --requires=wayland-protocols/1.31 -b missing
```

### Logs

```
wayland-protocols/1.31: RUN: meson setup --native-file "/home/ecko/.conan2/p/t/wayla2056c69a630b1/b/build-release/conan/conan_meson_native.ini" "/home/ecko/.conan2/p/t/wayla2056c69a630b1/b/build-release" "/home/ecko/.conan2/p/t/wayla2056c69a630b1/b/src" -Dprefix="/home/ecko/.conan2/p/t/wayla2056c69a630b1/p"
The Meson build system
Version: 1.0.0
Source dir: /home/ecko/.conan2/p/t/wayla2056c69a630b1/b/src
Build dir: /home/ecko/.conan2/p/t/wayla2056c69a630b1/b/build-release
Build type: native build
Project name: wayland-protocols
Project version: 1.31
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring wayland-protocols.pc using configuration
Configuring wayland-protocols-uninstalled.pc using configuration
Build targets in project: 0

wayland-protocols 1.31

  User defined options
    Native files: /home/ecko/.conan2/p/t/wayla2056c69a630b1/b/build-release/conan/conan_meson_native.ini
    prefix      : /home/ecko/.conan2/p/t/wayla2056c69a630b1/p


ERROR: Could not detect Ninja v1.8.2 or newer

A full log can be found at /home/ecko/.conan2/p/t/wayla2056c69a630b1/b/build-release/meson-logs/meson-log.txt

wayland-protocols/1.31: ERROR:
Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' build failed
```

It seems that `meson` can't find `ninja` even though both are fetched from Conan.
Having both installed via `apt` or via `pip3` doesn't make any difference.

**Comment by memsharded** [comment]
Hi @mkmkme 

Can you please post the full output that shows the actual dependencies used?
This could be a recipe issue, that something is not fully ready in some of those recipes for 2.0.

**Comment by memsharded** [comment]
I can confirm that a basic:
- ``conan new meson_lib -d name=pkg -d version=0.1
- add ``tool_requires = "meson/1.0.0``
- ``conan create .``

Fails with (not same issue, but it evidences that those packages might not be 2.0 ready):
```
conanvcvars.bat: Activating environment Visual Studio 16 - amd64 - vcvars_ver=14.2
[vcvarsall.bat] Environment initialized for: 'x64'
'meson' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by memsharded** [comment]
Testing things, I have realized there is a bug in ``MesonToolchain``, that makes necessary to explicitly instantiate ``VirtualBuildEnv`` in recipes, for environment to work. I have submitted https://github.com/conan-io/conan/pull/13346 for 2.0.2 to fix it.

With this fix the basic:
- ``conan new meson_lib -d name=pkg -d version=0.1``
- Creating a profile with ``[tool_requires]`` ``meson/1.0.0`` and ``pkgconf/1.9.3``
- ``conan create . -pr=profile``

Next step is to check the specifics of the wayland recipe.

**Comment by SpaceIm** [comment]
The recipe is also a little bit wrong, it calls VirtualBuildEnv after MesonToolchain.

**Comment by memsharded** [comment]
This was closed by #13346, it seems that from the Conan client side the MesonToolchain would be working fine (when 2.0.2 is released). I'd suggest re-checking when this is released, and if the ``wayland`` recipe still fails, open a ticket in ``conan-center-index`` repo to check potential issues with the recipe.
